earn/src/data/Uniswap.ts

```diff
@@ -98,3 +121,111 @@ export function getValueOfLiquidity(position: UniswapPosition, currentTick: numb
 
   return new Big(value.toString(10)).div(10 ** token1Decimals).toNumber();
 }
+
+export function uniswapPositionKey(owner: string, lower: number, upper: number): string {
+  return ethers.utils.solidityKeccak256(['address', 'int24', 'int24'], [owner, lower, upper]);
+}
+
+export function convertSqrtPriceX96(sqrtPriceX96: ethers.BigNumber): Big {
+  const priceX96 = sqrtPriceX96.mul(sqrtPriceX96).div(Q96);
+  return toBig(priceX96).div(BIGQ96);
+}
+
+export function tickToPrice(
+  tick: number,
+  token0Decimals: number,
+  token1Decimals: number,
+  isInTermsOfToken0 = true
+): number {
+  const sqrtPriceX96 = TickMath.getSqrtRatioAtTick(tick);
+  const priceX192 = JSBI.multiply(sqrtPriceX96, sqrtPriceX96);
+  const priceX96 = JSBI.signedRightShift(priceX192, JSBI.BigInt(96));
+
+  const priceX96Big = new Big(priceX96.toString(10));
+
+  const decimalDiff = token0Decimals - token1Decimals;
+  const price0In1 = priceX96Big
+    .mul(10 ** decimalDiff)
+    .div(BIGQ96)
+    .toNumber();
+  const price1In0 = 1.0 / price0In1;
+  return isInTermsOfToken0 ? price0In1 : price1In0;
+}
```

Comment:
I think we have functions like this in BalanceSheet already. Maybe we should create a file just for tick/sqrt/price conversions

Reply:
In prime these are both in Uniswap.ts. While I am not against moving this stuff to a separate file it is out of scope of this PR